pkg/cmd/experimental/syncgroups/ad/ldapinterface.go

```diff
@@ -73,15 +73,17 @@ func (e *ADLDAPInterface) ExtractMembers(ldapGroupUID string) ([]*ldap.Entry, er
 			return nil, err
 		}
 
-		for i := range currEntries {
-			currEntry := currEntries[i]
```

Comment:
No.  This will break us.  The code here makes sure that you add a different pointer to the list for every entry.  The change you just made will assign them all to the same pointer, because `currEntry` in your code is a single spot in memory.